src/main/java/com/stripe/model/BalanceTransactionDeserializer.java

```diff
@@ -0,0 +1,76 @@
+package com.stripe.model;
+
+import java.lang.reflect.Type;
+
+import com.google.gson.FieldNamingPolicy;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonPrimitive;
+
+public class BalanceTransactionDeserializer implements JsonDeserializer<BalanceTransaction> {
+
+	@Override
+	public BalanceTransaction deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
+		Gson gson = new GsonBuilder()
+				.setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
+				.create();
```

Comment:
So, I actually tried to do it this way initially, but `BalanceTransactionTest` was failing with `java.lang.StackOverflowError` in that case... Not sure why exactly, but APIResource.GSON just doesn't work there, while new Gson instance does.